File: server/src/routes/Optimize.js
const express = require('express');
const router = express.Router();

const optimizeController = require('../controllers/OptimizeController');

router.post('/goldenSectionSearch', optimizeController.goldenSectionSearch);
router.post('/parabolicInterpolation', optimizeController.parabolicInterpolation);
router.post('/newtonMethod', optimizeController.newtonMethod);

module.exports = router;

File: flask-server/routes/Optimize.py
from flask import Blueprint

from controllers.OptimizationController import OptimizationController

Optimize = Blueprint("Optimize", __name__)

Optimize.route("/newtonMethod", methods=["POST"])(OptimizationController.newtonMethod)
Optimize.route("/goldenSectionSearch", methods=["POST"])(OptimizationController.goldenSectionSearch)

